### PR TITLE
Add an extra grace period for new connections

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

We've seen felix get cyclicly booted from the server, presumably
because sending the snapshot took an unexpectedly long time due
to CPU or netowrk contention.

Add an extra grace period for newly-connected clients so that
they get some time to catch up after the snapshot is sent. Hopefully,
that will solve the problem outright but it also has the side effect
of giving the new felix some time to update the dataplane before we
kill the Typha connection.  This ensures that it makes progress
even if it's restarting.

## Todos
- [x] Unit tests (full coverage)
- [x] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
Typha now gives newly connected clients an extra grace period to catch up after sending the snapshot.  Should reduce the possibility of cyclic disconnects.
```

### CODE REVIEW

File: pkg/syncserver/sync_server.go
Comment (robbrockbank) [inline pkg/syncserver/sync_server.go:788]: nit: progresseven

**Review by robbrockbank — APPROVED** [review]
LGTM - just a thought - it might be nice to have some prometheus stats for this.

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
